docs/migrate-to-antrea.md

```diff
@@ -0,0 +1,69 @@
+# Migrate from other CNI to Antrea
+
+This document describes how to migrate from other CNIs to Antrea.
+
+NOTE: The following is a reference list of CNIs and versions we have already
+verified the migration process. CNIs and versions that are not listed here
+might also work, please create an issue if you run into problems during the
+migration to Antrea. In addition, during the migration process, no Kubernetes
+resources should be created or deleted, otherwise the migration process might
+fail or some unexpected problems might occur.
+
+| CNI     | Version |
+|---------|---------|
+| Calico  | v3.26   |
+| Flannel | v0.22.0 |
+
+With the help of migration [script](../hack/migrate-to-antrea.sh),
+the migration process is fully automated. You can use the following
+command to start the migration:
+
+```bash
+./hack/migrate-to-antrea.sh
+```
+
+More details about the migration process are described in the following
+sections. And you can learn how to use the script by running
+`./hack/migrate-to-antrea.sh --help`.
+
+The migration process is divided into two steps:
+
+1. Install Antrea in the cluster.
```

Comment:
We discussed this in https://github.com/antrea-io/antrea/pull/5677#discussion_r1424042334.
I still think there is no need to repeat the installation process and doc here. There are plenty of ways of installing antrea in various platforms. **I don't think just supporting one of the methods and asking users to download the manifest locally and help them apply the manifest via this script is really helpful.**  Why can't they just apply the manifest themselves or choose other way of installing?

And applying the default manifest which points to the development branch is not something we really recommend.